test/Microsoft.AspNet.Mvc.Xml.Test/XmlAssert.cs

```diff
@@ -20,21 +22,47 @@ public static class XmlAssert
         /// <param name="actualXml">Actual xml string.</param>
         public static void Equal(string expectedXml, string actualXml)
         {
-            var sortedExpectedXDoc = SortAttributes(XDocument.Parse(expectedXml));
-            var sortedActualXDoc = SortAttributes(XDocument.Parse(actualXml));
+            if (string.Equals(expectedXml, actualXml, StringComparison.Ordinal))
+            {
+                return;
+            }
+
+            var sortedExpectedXDocument = SortAttributes(XDocument.Parse(expectedXml));
+            var sortedActualXDocument = SortAttributes(XDocument.Parse(actualXml));
 
-            bool areEqual = XNode.DeepEquals(sortedExpectedXDoc, sortedActualXDoc);
+            // Since XNode's DeepEquals does not check for presence of xml declaration,
+            // check it explicitly
+            bool areEqual = EqualDeclarations(sortedExpectedXDocument.Declaration, sortedActualXDocument.Declaration);
+
+            areEqual = areEqual && XNode.DeepEquals(sortedExpectedXDocument, sortedActualXDocument);
 
             if (!areEqual)
             {
-                throw new EqualException(sortedExpectedXDoc, sortedActualXDoc);
+                throw new EqualException(sortedExpectedXDocument.GetRawXml(), sortedActualXDocument.GetRawXml());
+            }
+        }
+
+        private static bool EqualDeclarations(XDeclaration expected, XDeclaration actual)
+        {
+            if (expected == null && actual == null)
+            {
+                return true;
             }
+
+            if (expected == null || actual == null)
+            {
+                return false;
+            }
+
+            return string.Equals(expected.Version, actual.Version, StringComparison.OrdinalIgnoreCase)
+                && string.Equals(expected.Encoding, actual.Encoding, StringComparison.OrdinalIgnoreCase);
         }
 
         private static XDocument SortAttributes(XDocument doc)
         {
             return new XDocument(
                     doc.Declaration,
```

Comment:
I need this as I parse the xml strings to get the declarations and then do the comparison on them.